File: problems/121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxProfit = 0
        l = 0
        for r in range(len(prices)):
            if prices[r] > prices[l]:
                maxProfit = max(maxProfit, prices[r] - prices[l])
            else:
                l = r
        return maxProfit

File: problems/121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
const maxProfit = (prices) => {
    let l = 0;
    let r = 1;
    let maxProfit = 0;
    while (r < prices.length) {
        if (prices[l] < prices[r]) {
            maxProfit = Math.max(maxProfit, prices[r] - prices[l]);
        } else {
            l = r;
        }
        r ++;
    }
    return maxProfit;
};